### PR TITLE
[image][online_image][animation] Fix transparency in RGB565

### PR DESCRIPTION
**Pull request by clydebarrow**
# What does this implement/fix?

When images were converted to RGB565 with transparency, especially from SVG, anti-aliasing was lost as the image format use chroma-keying instead of an alpha channel.

This PR changes the RGB565 with transparency format to one that has a proper alpha channel, so anti-aliasing works as expected.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

**Related issue or feature (if applicable):** fixes
* https://github.com/esphome/issues/issues/6047
* https://discord.com/channels/429907082951524364/1297013896099532810

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [x] ESP32
- [x] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx
- [x] host

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/image/image.h [hunk 1]
```diff
@@ -3,12 +3,7 @@
 #include "esphome/components/display/display.h"
 
 #ifdef USE_LVGL
-// required for clang-tidy
-#ifndef LV_CONF_H
-#define LV_CONF_SKIP 1  // NOLINT
-#endif                  // LV_CONF_H
-
-#include <lvgl.h>
+#include "esphome/components/lvgl/lvgl_proxy.h"
```
Comment (guillempages) [inline esphome/components/image/image.h]: Shouldn't this part be a separate PR?
one PR to fix the transparency, and one PR to fix the host compilation?

File: esphome/components/image/image.h [hunk 2]
```diff
@@ -49,6 +26,25 @@ class Image : public display::BaseImage {
   const uint8_t *get_data_start() const { return this->data_start_; }
   ImageType get_type() const;
 
+  int get_bpp() const {
```
Comment (guillempages) [inline esphome/components/image/image.h:29]: Having this as a static method (as it was), is slightly better performant...
It's not a big deal, but I've done some tests with a (very big -> 550x800) image, and I'm getting 4 ms difference in rendering.

With this PR it's averaging 1.486s per frame; without it was taking 1.482s.
As I said; not much, but since you are always trying to squeeze the most performance on graphics, it might be worth a second look...

File: esphome/components/online_image/online_image.h [hunk 3]
```diff
@@ -86,13 +86,9 @@ class OnlineImage : public PollingComponent,
   Allocator allocator_{Allocator::Flags::ALLOW_FAILURE};
 
   uint32_t get_buffer_size_() const { return get_buffer_size_(this->buffer_width_, this->buffer_height_); }
-  int get_buffer_size_(int width, int height) const {
-    return std::ceil(image::image_type_to_bpp(this->type_) * width * height / 8.0);
-  }
+  int get_buffer_size_(int width, int height) const { return (this->get_bpp() * width + 7u) / 8u * height; }
```
Comment (guillempages) [inline esphome/components/online_image/online_image.h:89]:
```suggestion
  int get_buffer_size_(int width, int height) const { return (this->get_width_stride() * height; }
```

Comment (guillempages) [reply esphome/components/online_image/online_image.h:89]: Oh, no, the suggestion doesn't work, since it is not using the member width_ :-(

**Review by guillempages — COMMENTED** [review]
Hm. I don't know why, but when testing this PR with an animation, the animation is now showing wrong.
Looks like there is something wrong with reading the pixels. (My animation is using RGB565 with transparency; maybe the drawing part of the animation also needs updating for the 3bytes per pixel somewhere?

Took a video: https://github.com/user-attachments/assets/232af92a-6b51-43ca-8dc2-eed7aaec9a21

Without the pr:
https://github.com/user-attachments/assets/199408d7-12ef-4ef3-9221-af625bc92340


(Ignore the cross/"HI" on the top left, that's a different issue (I hope))

**Comment by clydebarrow** [comment]
> maybe the drawing part of the animation also needs updating for the 3bytes per pixel somewhere?

More than likely. I did make some changes to animation, but didn't test it :thinking:  I did test regular images and online images. Will do some testing and update.

File: esphome/components/image/image.h (hunk 1 above)
Comment (clydebarrow) [reply esphome/components/image/image.h]: Perhaps, but I needed that for testing anyway.

File: esphome/components/image/image.h (hunk 2 above)
Comment (clydebarrow) [reply esphome/components/image/image.h:29]: I'll have a look.

Comment (clydebarrow) [reply esphome/components/image/image.h:29]: I don't think `get_bpp()` has anything to do with it, it's not called in a loop anywhere that I can see. And that's a 0.25% difference, that's in the range of measurement error. You know that using LVGL to draw images is at least twice as fast as the display lambda?

**Comment by clydebarrow** [comment]
> but when testing this PR with an animation, the animation is now showing wrong.

Fixed now - the python code to convert the animated GIF needed updating too.

**Review by jesserockz — APPROVED** [review]
